### PR TITLE
Initial implementation of performance schema.

### PR DESCRIPTION
**Pull request by pixlwave**
This PR is likely for discussion right now. It is based on the existing iOS performance metrics that we were tracking in Matomo (and would like to continue tracking as these are helpful to look at when users report issues). I have the following questions:

- Should there be a single `Performance` event that has fields for `operation` and `category` rather than how I've done in this initial implementation where I made an event per category? Or it could event be a single event where the operation combines the two e.g. `StartupMountData`.
- We currently track duration in seconds, but in this PR I've used milliseconds to match the duration of the call event. Looking at the graphs, I wonder if seconds would actually be more obvious to read/interpret?
- I imagine these metrics can be useful on Android too, but it's worth discussing if any of these enum values should be prefixed like some of the screen names.

Note: I've added an iOS Startup Performance graph to the test [dashboard](https://posthog-poc.lab.element.dev/dashboard/1) to learn how this data is filtered and presented.

### CODE REVIEW

**Comment by novocaine** [comment]
Nice! This is exactly the sort of thing we should be tracking; posthog can give us percentiles for this. 

I think I'd like one way of representing performance metrics .. suggestion would be an event called PerformanceTimer, which has a name enum and a time in ms property

For other ideas, stats defines some standard ways of representing performance counters e.g. see https://statsd.readthedocs.io/en/v3.3/types.html

**Comment by novocaine** [comment]
> * We currently track duration in seconds, but in this PR I've used milliseconds to match the duration of the call event. Looking at the graphs, I wonder if seconds would actually be more obvious to read/interpret?

I think I'd prefer integer ms to float seconds? But I don't feel strongly about it.

**Comment by pixlwave** [comment]
@novocaine Updated with your comments, `PerformanceTimer` makes much more sense to me.

**Review by bmarty — APPROVED** [review]
LGTM. I'm wondering if we can add doc on the enum value, for instance what is measured for `StartupIncrementalSync`?

File: schemas/PerformanceTimer.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
+  "type": "object",
+  "description": "Triggered after timing a long running operation in the app.",
```
Comment (novocaine) [inline schemas/PerformanceTimer.json:3]: doesn't have to be long running!

**Review by novocaine — APPROVED** [review]

**Comment by novocaine** [comment]
:shipit:

**Pull request merged**